content.go

```diff
@@ -60,14 +63,25 @@ type ResolveOptions struct {
 }
 
 // Resolve resolves a descriptor with provided reference from the target.
-func Resolve(ctx context.Context, target ReadOnlyTarget, ref string, opts ResolveOptions) (ocispec.Descriptor, error) {
+func Resolve(ctx context.Context, target ReadOnlyTarget, reference string, opts ResolveOptions) (ocispec.Descriptor, error) {
 	if opts.TargetPlatform == nil {
-		return target.Resolve(ctx, ref)
+		return target.Resolve(ctx, reference)
+	}
+
+	proxy := cas.NewProxy(target, cas.NewMemory())
```

Comment:
Why do we need a cache proxy here for all code path?

Reply:
Nice catch!